webportal/src/app/market_list/components/side_bar.jsx

```diff
@@ -1,59 +1,68 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
-import React, { useContext, useState } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { getTheme, Stack, Text, SearchBox } from 'office-ui-fabric-react';
-import queryString from 'query-string';
-import { isNil } from 'lodash';
-import TypeFilter from './type_filter';
+import { isEmpty } from 'lodash';
+import FilterItem from './type_filter';
 import Context from 'App/context';
 import { TYPE_ENUM } from 'App/utils/constants';
+import PropTypes from 'prop-types';
 
-const SideBar = () => {
-  const [type, setType] = useState(null);
-  const [keyword, setKeyword] = useState();
+const SideBar = props => {
+  const { categories } = props;
+  const [currentType, setCurrentType] = useState(null);
+  const [currentAuthor, setCurrentAuthor] = useState(null);
+  const [currentCategory, setCurrentCategory] = useState(null);
   const { spacing } = getTheme();
-  const { history, user } = useContext(Context);
+  const { history, user, routeProps } = useContext(Context);
 
-  const getQueryString = (type, keyword) => {
-    let qs = '';
-    if (isNil(keyword) || keyword === '') {
-      if (isNil(type)) {
-        qs = '';
-      } else if (type === 'my') {
-        qs = queryString.stringify({ author: user });
-      } else {
-        qs = queryString.stringify({ type });
-      }
-    } else {
-      if (isNil(type)) {
-        qs = queryString.stringify({ keyword });
-      } else if (type === 'my') {
-        qs = queryString.stringify({ author: user, keyword });
-      } else {
-        qs = queryString.stringify({ type, keyword });
-      }
-    }
-    return qs;
-  };
+  useEffect(() => {
+    const searchParams = new URLSearchParams(routeProps.location.search);
+    setCurrentType(searchParams.get('type'));
+    setCurrentAuthor(searchParams.get('author'));
+    setCurrentCategory(searchParams.get('category'));
+  }, []);
 
-  const changeType = newType => {
+  const changeUrl = (section, value) => {
     return () => {
-      if (type === newType) {
-        setType(null);
-        const qs = getQueryString(null, keyword);
-        history.push(`/?${qs}`);
-      } else {
-        setType(newType);
-        const qs = getQueryString(newType, keyword);
-        history.push(`/?${qs}`);
+      const searchParams = new URLSearchParams(routeProps.location.search);
+      if (section === 'keyword') {
+        if (value === '') {
+          searchParams.delete('keyword');
+        } else {
+          searchParams.set('keyword', value);
+        }
+      } else if (section === 'type') {
+        if (currentType === value) {
+          setCurrentType(null);
+          searchParams.delete('type');
```

Comment:
a little confused: so why do we remove the type from search params. I can understand that when the keyword is null, you will remove it, but why not keep the value in searchParams?

Reply:
You mean like `searchParams.set('type', null);`? Here the del action is triggered when inverse selection happens. So we should delete the previous type value. Set type to null also works but the url will look weired like `xxx?type=null`.